monkey/infection_monkey/network_scanning/mssql_fingerprinter.py

```diff
@@ -77,27 +80,51 @@ def _query_mssql_for_instance_data(host: str) -> Optional[bytes]:
         sock.close()
 
 
-def _get_services_from_server_data(data: bytes) -> Dict[str, Any]:
-    services = {MSSQL_SERVICE: {}}
-    services[MSSQL_SERVICE]["display_name"] = DISPLAY_NAME
-    services[MSSQL_SERVICE]["port"] = SQL_BROWSER_DEFAULT_PORT
+def _get_instance_info(instance_info: Sequence[str]) -> Dict[str, str]:
+    info = {}
+    it = iter(instance_info)
+    for k in it:
+        info[k] = next(it)
+
+    return info
+
+
+def _parse_instance(instance: str) -> Dict[str, str]:
+    instance_info = instance.split(";")
+    if len(instance_info) > 1:
+        return _get_instance_info(instance_info)
+
+    return {}
+
+
+def _get_services_from_server_data(data: bytes) -> List[DiscoveredService]:
+    services = []
+    discovered_network_service = NetworkService.UNKNOWN
 
     # Loop through the server data
+    # Example data:
+    # yServerName;MSSQL-16;InstanceName;MSSQLSERVER;IsClustered;No;Version;14.0.1000.169;tcp;1433;np;\\MSSQL-16\pipe\sql\query;;
     mssql_instances = filter(lambda i: i != "", data[3:].decode().split(";;"))
-
     for instance in mssql_instances:
-        instance_info = instance.split(";")
-        if len(instance_info) > 1:
-            services[MSSQL_SERVICE][instance_info[1]] = {}
-            for i in range(1, len(instance_info), 2):
-                # Each instance's info is nested under its own name, if there are multiple
-                # instances
-                # each will appear under its own name
-                services[MSSQL_SERVICE][instance_info[1]][instance_info[i - 1]] = instance_info[i]
-
-            logger.debug(f"Found MSSQL instance: {instance}")
-
-    if len(services[MSSQL_SERVICE].keys()) == 2:
-        services = {}
+        instance_info = _parse_instance(instance)
+        if "tcp" in instance_info:
```

Comment:
```suggestion
        if "tcp" in instance.split(";"):
```
And that's it? Why do we need those `_parse_instance` and `_get_instance_info`?

Reply:
Because we create a dictionary of the instance info for later when we take the port out of it.

Reply:
We don't need `instance_info` for later, because `instance_info["tcp"]` is just "tcp" afaik. Or the type hint in `get_instance_info(instance_info: Sequence[str])` is wrong

Reply:
Ahh, ok I see what's being done. I'll add a note at the relevant line